core/src/mindustry/input/InputHandler.java

```diff
@@ -1161,7 +1163,15 @@ void iterateLine(int startX, int startY, int endX, int endY, Cons<PlaceLine> con
         }
 
         if(diagonal){
-            points = Placement.pathfindLine(block != null && block.conveyorPlacement, startX, startY, endX, endY);
+            Tile start = world.tile(startX, startY);
+            Tile end = world.tile(endX, endY);
+            if(block != null && block instanceof Conveyor && start.block() instanceof Conveyor && end.block() instanceof Conveyor && start.block() != block){
```

Comment:
Conduits/conveyors should not be hardcoded into the system with special functions to handle them. The logic for each type of block is practically identical.

Reply:
I really don't know how to put them under one roof. Perhaps do something with AutoTiler?

Reply:
add it to an interface